core/src/avm2/namespace.rs

```diff
@@ -336,3 +336,77 @@ impl<'gc> Namespace<'gc> {
         }
     }
 }
+
+/// List of common namespaces used in the rest of the AVM2.
+#[derive(Collect)]
+#[collect(no_drop)]
+pub struct CommonNamespaces<'gc> {
+    public_namespaces: [Namespace<'gc>; CommonNamespaces::PUBLIC_LEN],
+
+    pub(super) internal: Namespace<'gc>,
+    pub(super) as3: Namespace<'gc>,
+    pub(super) vector_public: Namespace<'gc>,
+    pub(super) vector_internal: Namespace<'gc>,
+    pub(super) proxy: Namespace<'gc>,
+
+    // These are required to facilitate shared access between Rust and AS.
+    pub(super) flash_display_internal: Namespace<'gc>,
+    pub(super) flash_utils_internal: Namespace<'gc>,
+    pub(super) flash_geom_internal: Namespace<'gc>,
+    pub(super) flash_events_internal: Namespace<'gc>,
+    pub(super) flash_text_engine_internal: Namespace<'gc>,
+    pub(super) flash_net_internal: Namespace<'gc>,
+
+    pub(super) __ruffle__: Namespace<'gc>,
```

Comment:
Can this be called `ruffle_internal` instead?

Reply:
This namespace is actually public, because it's purpose for Ruffle-specific functions we want to call in tests.